mantidproject/mantid
# Add sparse workspace features support to the indirect correction UI

**Description of work.**
This PR adds sparse workspace features support to the "Calculate Monte Carlo Absorption" tab in the Indirect Corrections UI. Three fields have been added to the UI spare instrument checkbox, the number of detector rows and the number of detector columns. Note, that these fields are already present in the PaalmanPingsMonteCarloAbsorption algorithm UI, so I just added them to the correction UI.


**Report to:** Spencer Howells

**To test:**

From the top bar open Interfaces->Indirect->Corrections. Then open Calculate Monte Carlo Absorption tab. You will find the new fields added in the Monte Carlo section. After that, select a sample workspace and input some values in the new fields then click run. Finally, check the history of the output to ensure the fields are recorded in the history of the algorithms. 

Fixes #34716. 



*This does not require release notes* because **it is a simple UI modification**



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
